src/macros.jl

```diff
@@ -231,7 +231,35 @@ function extract_kwargs(args)
     return flat_args, kwargs, requestedcontainer
 end
 
-function addkwargs!(call, kwargs)
+"""
+    add_kwargs(call, kwargs)
+
+Add the keyword arguments `kwargs` to the function call expression `call`,
+escaping the expressions. The keyword arguments can be obtained with
+[`extract_kwargs`](@ref).
+
+## Examples
+
+```jldoctest; setup = :(using JuMP)
+julia> call = :(f(1, a=2))
+:(f(1, a=2))
+
+julia> using JuMP
+
+julia> macro m(args...)
+           flag_args, kwargs, requested_container
+           JuMP.extract_kwargs(args); JuMP.add_keyword_arguments(call, kwargs)
```

Comment:
It would be sufficient to provide a standalone example without macros. There's a lot of nontrivial things going on in this example (referring to an object in global scope, using `extract_kwargs`. If you provide a more complex example then it needs more explanation.